documentation/puppet-api/v3/environment_modules.json

```diff
@@ -0,0 +1,26 @@
+{
+  "$schema":     "http://json-schema.org/draft-04/schema#",
+  "title":       "Environment Modules",
+  "description": "Information about the modules in a Puppet code environment",
+  "type":        "object",
+  "properties": {
+    "modules": {
+      "description": "The array of modules which exist in an environment.",
+      "type": "array",
+      "properties": {
+        "name": {
+          "description": "The name of the puppet module",
+          "type": "string"
+        },
+        "version": {
+          "description": "The version of the puppet module",
+          "type": "string"
+        }
+      }
+    },
+    "name": {
+      "description": "Name of the environment",
+      "type": "string"
+    }
+  }
+}
```

Comment:
I think this schema isn't strict enough to catch malformed json.  I used the ruby `json-schema` gem to validate JSON responses that I captured from the endpoint.  Here's the command line I ran for that:

```
ruby -rjson-schema -e "puts JSON::Validator.validate!('./documentation/puppet-api/v3/environment_modules_fixed.json','response.json')"
```
'response.json' is the file I captured from a curl request to the endpoint.  Even if I intentionally messed up some of the key names, e.g., replacing "name" with "bogus-name", the `validate!` call would still return `true`.

I added a couple of `required` elements with specific names expected in the output and added an `items` wrapper underneath the `array` element, which I think is required for JSON schema validation of an array.  The result is below.  With this, I saw validation failures for any intentionally malformed json vs. success for the "good" json that the endpoint returned.

Does this look okay to you?

```json
{
  "$schema":     "http://json-schema.org/draft-04/schema#",
  "title":       "Environment Modules",
  "description": "Information about the modules in a Puppet code environment",
  "type":        "object",
  "properties": {
    "modules": {
      "description": "The array of modules which exist in an environment.",
      "type": "array",
      "items": {
        "type": "object",
        "properties": {
          "name": {
            "description": "The name of the puppet module",
            "type": "string"
          },
          "version": {
            "description": "The version of the puppet module",
            "type": "string"
          }
        },
        "required": ["name", "version"],
        "additionalProperties": false
      }
    },
    "name": {
      "description": "Name of the environment",
      "type": "string"
    }
  },
  "required": ["modules", "name"],
  "additionalProperties": false
}
```